app/controllers/concerns/repo_setup.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module RepoSetup
+  extend ActiveSupport::Concern
+
+  def setup_status(assignment_repo)
+    repo     = assignment_repo.github_repository
+    progress = { status: :importing }
+
+    return progress unless repo.import_progress[:status] == "complete"
+
+    progress.update(status: :configuring) if assignment_repo.configuring?
+    progress.update(status: :complete) if assignment_repo.configured? || !repo.branch_present?(config_branch)
```

Comment:
Do you have any recommendations for this? Would it be better to have a method under GitHubClassroom for it instead?